.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,34 @@
+name: Build and Deploy Docs
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  build-publish:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+          cache: yarn
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Build website
+        run: yarn build
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
+        if: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
Needed?

Reply:
Since we're running the pipeline to build a PR, but only want to deploy when merged to `main`, that's why this is here

Reply:
You can control that from the trigger at the beginning of the file.

Reply:
Indeed, but they are not equivalent. 
* Control from triggers at the top of the file => The whole file only runs on merge to main. On PRs nothing runs

Current behavior:
* on merge to main: build + deploy
* on PR: build only

I would say this (Always build, but only deploy when merged to main) is standard behavior for many pipelines

Reply:
We can also to 2 files based on triggers like `on-push.yml` and `on-merge.yml`. There would be a bit of duplication. Happy with your current setup anyway.